grafana-plugin/src/GrafanaPluginRootPage.tsx

```diff
@@ -150,12 +151,20 @@ export const Root = observer((props: AppRootProps) => {
     onNavChanged(navModel);
   }, [navModel, onNavChanged]);
 
-  const Page = pages.find(({ id }) => id === page)?.component || pages[0].component;
+  const { action: pagePermissionAction, component: PageComponent } = pages.find(({ id }) => id === page) || pages[0];
+  const userHasAccess = pagePermissionAction ? !!grafanaUser.permissions[pagePermissionAction] : true;
 
   return (
     <DefaultPageLayout {...props}>
       <GrafanaTeamSelect currentPage={page} />
-      <Page {...props} path={pathWithoutLeadingSlash} />
+      {userHasAccess ? (
+        <PageComponent {...props} path={pathWithoutLeadingSlash} />
+      ) : (
+        <>
+          {/* TODO: what should be the behavior here? a redirect? if so, to what page? or maybe an unauthorized component? */}
+          <p>You don't have access!</p>
```

Comment:
my thoughts here are that we could either redirect the user to the first page that they have access to, or if they don't have access to any, redirect them to the main grafana dashboard page?